From dff2a8703cef984fe56285fb3afcbe04f12bc006 Mon Sep 17 00:00:00 2001
From: Ralf Gommers <ralf.gommers@gmail.com>
Date: Tue, 6 Sep 2022 12:55:57 +0300
Subject: [PATCH] BLD: fix usage of `get_install_data`, which defaults to
 purelib

This resulted in `{py_purelib}/` entries in
`<build-dir>/meson-info/intro-install_plan.json`, and there should
be zero of those.

[skip azp] [skip circle]
---
 meson.build               | 2 ++
 scipy/linalg/meson.build  | 6 +++---
 scipy/meson.build         | 4 ++--
 scipy/special/meson.build | 4 ++--
 4 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index fee19d759e400eed8a61d724e7702ac4ab92122c..759e6156966e3684d50a3e0fedca5edcbd517014 100644
--- a/meson.build
+++ b/meson.build
@@ -66,6 +66,8 @@ copier = find_program(['cp', 'scipy/_build_utils/copyfiles.py'])
 
 # https://mesonbuild.com/Python-module.html
 py_mod = import('python')
+# NOTE: with Meson >=0.64.0 we can add `pure: false` here and remove that line
+# everywhere else, see https://github.com/mesonbuild/meson/pull/10783.
 py3 = py_mod.find_installation()
 py3_dep = py3.dependency()
 
diff --git a/scipy/linalg/meson.build b/scipy/linalg/meson.build
index b885b224ef360e220aec73d712f2a9faf24e64d6..81ea88b47b0f29af74ede9ee516c42a8af6b0047 100644
--- a/scipy/linalg/meson.build
+++ b/scipy/linalg/meson.build
@@ -28,7 +28,7 @@ cython_linalg = custom_target('cython_linalg',
   # TODO - we only want to install the .pxd files! See comments for
   #        `pxd_files` further down.
   install: true,
-  install_dir: py3.get_install_dir() / 'scipy/linalg'
+  install_dir: py3.get_install_dir(pure: false) / 'scipy/linalg'
 )
 
 # pyx -> c, pyx -> cpp generators, depending on __init__.py here.
@@ -322,7 +322,7 @@ py3.install_sources(
 #       https://mesonbuild.com/Installing.html says is for build targets to
 #       use:
 #         `custom_target(..., install: true, install_dir: ...)
-#         # should use `py3.get_install_dir() / 'scipy/linalg'` ?
+#         # should use `py3.get_install_dir(pure: false) / 'scipy/linalg'` ?
 #       see https://github.com/mesonbuild/meson/issues/3206
 #
 #       For the below code to work, the script generating the files should use
@@ -340,7 +340,7 @@ py3.install_sources(
 #  output : ['cython_blas2.pxd', 'cython_lapack2.pxd'],
 #  command : ['cp', '@INPUT0@', '@OUTPUT0@', '&&', 'cp', '@INPUT1@', '@OUTPUT1@'],
 #  install : true,
-#  install_dir: py3.get_install_dir() / 'scipy/linalg'
+#  install_dir: py3.get_install_dir(pure: false) / 'scipy/linalg'
 #)
 
 subdir('tests')
diff --git a/scipy/meson.build b/scipy/meson.build
index 4a0274b491db05c94a76f65ebb3bf889f6a609db..eff65facfb551c6271d0ad4b2d3b1e55c447b7b9 100644
--- a/scipy/meson.build
+++ b/scipy/meson.build
@@ -158,7 +158,7 @@ generate_config = custom_target(
   output: '__config__.py',
   input: '../tools/config_utils.py',
   command: [py3, '@INPUT@', '@OUTPUT@'],
-  install_dir: py3.get_install_dir() / 'scipy'
+  install_dir: py3.get_install_dir(pure: false) / 'scipy'
 )
 
 generate_version = custom_target(
@@ -169,7 +169,7 @@ generate_version = custom_target(
   output: 'version.py',
   input: '../tools/version_utils.py',
   command: [py3, '@INPUT@', '--source-root', '@SOURCE_ROOT@'],
-  install_dir: py3.get_install_dir() / 'scipy'
+  install_dir: py3.get_install_dir(pure: false) / 'scipy'
 )
 
 python_sources = [
diff --git a/scipy/special/meson.build b/scipy/special/meson.build
index bd541d722108559daf18ffdcf0fe12ff29643ae8..f1c954e804a7729e8f17db70ae73618718f7d192 100644
--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -341,7 +341,7 @@ cython_special = custom_target('cython_special',
   input: ['_generate_pyx.py', 'functions.json', '_add_newdocs.py'],
   command: [py3, '@INPUT0@', '-o', '@OUTDIR@'],
   install: true,
-  install_dir: py3.get_install_dir() / 'scipy/special'
+  install_dir: py3.get_install_dir(pure: false) / 'scipy/special'
 )
 
 # pyx -> c, pyx -> cpp generators, depending on copied pxi, pxd files.
@@ -486,7 +486,7 @@ foreach npz_file: npz_files
       '--use-timestamp', npz_file[2], '-o', '@OUTDIR@'
     ],
     install: true,
-    install_dir: py3.get_install_dir() / 'scipy/special/tests/data'
+    install_dir: py3.get_install_dir(pure: false) / 'scipy/special/tests/data'
   )
 endforeach
 
